types/chain.go

```diff
@@ -0,0 +1,65 @@
+package types
+
+import (
+	"errors"
+	"github.com/trustwallet/go-primitives/coin"
+)
+
+func GetChainFromAssetType(type_ string) (coin.Coin, error) {
```

Comment:
Avoid underscores `_` except for unit test. You call name it like `assetType` for example.